sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -335,6 +353,39 @@ public static Task<WorkspaceStack> CreateOrSelectStackAsync(LocalProgramArgs arg
 
         private static readonly string[] SettingsExtensions = new string[] { ".yaml", ".yml", ".json" };
 
+        private async Task GetPulumiVersionAsync(CancellationToken cancellationToken = default)
```

Comment:
Nit: `GetX` should return a value without side-effects. If we assign the version inside this method, I would call it `PopulatePulumiVersionAsync` or something along that line.